Lesson9/src/shop/Shop.java

```diff
@@ -0,0 +1,59 @@
+package shop;
+
+import exceptions.SearchProductException;
+import interfaces.IShop;
+import products.Product;
+
+import java.util.*;
+
+public class Shop implements IShop {
+    List<Product> productList = new ArrayList<>();
+
+    @Override
+    public void addProduct(Product product) throws SearchProductException {
+        if (ShopUtil.isTheProductPresent(productList, product.getId())) {
+            throw new SearchProductException(product.getName() + " is already in the list");
+        }
+        productList.add(product);
+    }
+
+    public void addProduct(List<Product> newList) throws SearchProductException {
+        for (Product product : productList) {
+            for (Product newProduct : newList) {
+                if (newProduct.getId() == (product.getId())) {
+                    throw new SearchProductException(newProduct.getName() + " is already in the list");
+                }
+            }
+        }
+        productList.addAll(newList);
```

Comment:
I'd replace it with
for (Product product: newList) {
addProduct(product);
}

in this case you will be 100% sure that all products are added In the same way

Reply:
If I replace it in the second for, I will not check condition with another products from the first for.

Reply:
Ha! I know decision! I need to swap both for